### PR TITLE
always remove tracktype if surface category changed 

### LINKED ISSUE #5951: Solving some track related quests can leave inconsistent data behind
Hi there,

some weeks ago I came across a situation where solving some simple quests turn consistent tagging into an inconsistent tag combination.

Consider the following situation: An unpaved agricultural track, tagged consistently with

```
highway=track
surface=compacted
tracktype=grade2
```

The smoothness quest popped up. However, this track has obviously been asphalted recently, so I answered with _"Other surface"_ from the _Äh..._ menu.

Now the surface quest popped up, which I answered with _"Asphalt"_.

Now the smoothness popped up again, which I answered with _"Excellent"_.

No further quests popped up, leaving the tagging as

```
highway=track
surface=asphalt
smoothness=excellent
tracktype=grade2
```

**Which is a contradiction** - I cannot imagine any case where this combination could be valid - tracks with "excellent" asphalt will **_always_** be "grade1" ([see the wiki](https://wiki.openstreetmap.org/wiki/Key:smoothness#Smoothness)).

Is there anything that could be improved here?

**Ideas:**

```
IF (highway="track" // or "path" or similar
    AND surface was changed from anything to "asphalt" // or "concrete" or "paving stones" or similar
    AND tracktype != "grade1") {
        delete tracktype
}
```
(SC users with activated "tracktype" quest would be able to correct this immediately, otherwise at least the now wrong "tracktype" tag would be deleted.)

Or maybe even better:

```
IF (highway="track"
    AND surface="asphalt"
    AND smoothness was changed from anything to "excellent"
    AND tracktype exists) {
        set tracktype to "grade1"
}
```

(which changes the tracktype to the only correct value but does not silently create this tag; if the latter is no problem then the last AND clause can be of course omitted.)

Observed with SC v58.

The cycling cat

### PR DESCRIPTION
**Pull request by westnordost**
also, remove confirmation message in the surface quest

* fixes #5951
* replaces and closes #5962

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/osm/surface/SurfaceUtils.kt [hunk 1]
```diff
@@ -40,42 +44,14 @@ val PAVED_SURFACES = FULLY_PAVED_SURFACES + setOf(
...
-private val SOFT_TRACK_SURFACES = setOf(
-    "earth", "dirt", "soil", "grass", "sand", "mud", "snow", "woodchips",
-)
-
+// very lenient, to not flag surface+trackype combinations as invalid in edge cases
 val INVALID_SURFACES_FOR_TRACKTYPES = mapOf(
-    "grade1" to SOFT_TRACK_SURFACES, // could be compacted, as long as it is "solid"
-    "grade2" to SOFT_TRACK_SURFACES,
+    "grade1" to SOFT_NATURAL_SURFACES, // could be compacted, as long as it is "solid"
+    "grade2" to SOFT_NATURAL_SURFACES,
```
[the remark this answers is not in the document]
Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/osm/surface/SurfaceUtils.kt:50]: kinda. I noticed it. It's not really relevant, IMO

File: app/src/test/java/de/westnordost/streetcomplete/osm/surface/SurfaceCreatorKtTest.kt [hunk 2]
```diff
@@ -45,39 +45,43 @@ class SurfaceCreatorKtTest {
...
             SurfaceAndNote(Surface.ASPHALT).appliedTo(mapOf(
+                "surface" to "compacted",
                 "tracktype" to "grade5",
                 "check_date:tracktype" to "2011-11-11"
             ))
         )
     }
 
-    @Test fun `remove mismatching tracktype not done with prefix`() {
+    @Test fun `don't remove tracktype when surface category didn't change`() {
```
[the remark this answers is not in the document]
Comment (westnordost) [reply app/src/test/java/de/westnordost/streetcomplete/osm/surface/SurfaceCreatorKtTest.kt:63]: okay

**Comment by westnordost** [comment]
Thanks for the review, Flo

**Comment by malteger** [comment]
This is destroying data. Please leave the tracktype inplace if the surface is changed from `unpaved` to a more specific but still unpaved surface type.

Currently, a change from unpaved to a 'natural' type is triggering a deletion of tracktype.

**Comment by matkoniecz** [comment]
What about say `surface=unpaved tracktype=grade1` changed to `surface=sand` ?

**Comment by mnalis** [comment]
- ref: https://github.com/streetcomplete/StreetComplete/issues/6258

**Comment by matkoniecz** [comment]
See also #6116 #5980 #5962 #5951 #5650 #5566 #5236 #5650 #5133 #4105 #1909 #662 

This surface-tracktype conflict handling is one of more annoying cases, where it is impossible to keep everyone happy and either way there will be some bad impact on OSM data. Either some valid `tracktype` will be removed or some `tracktype`conflicting with surface will be kept. Or interface will become horrifically complex and users will use this quest less.

**Pull request merged**